metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceUtil.java

```diff
@@ -140,9 +165,15 @@ public TablePartitionsUpdatePayload createTablePartitionsUpdatePayload(
      */
     public List<String> getSortedDeletionPartitionKeys(final List<PartitionDto> partitionDtos,
                                                        final String deleteColumn) {
-        final List<String> ret = partitionDtos.stream().map(
-            x -> PartitionUtil.getPartitionKeyValues(x.getName().toString()).get(deleteColumn)
-        ).sorted().collect(Collectors.toList());
+        //still check in case the function is used by other places
+        if (StringUtils.isEmpty(deleteColumn)) {
+            return Collections.emptyList();
+        }
+        final List<String> ret = partitionDtos.stream()
+            .map(x -> PartitionUtil.getPartitionKeyValues(x.getName().toString()).get(deleteColumn))
+            .filter(Objects::nonNull)
+            .sorted()
+            .collect(Collectors.toList());
         Collections.reverse(ret);
```

Comment:
You can also use sorted() method of streams.
..{
return partitionDtos.stream()
.map(x -> PartitionUtil.getPartitionKeyValues(x.getName().toString()).get(deleteColumn))
.filter(Objects::nonNull)
.sorted(Comparator.reversed())
.collect(Collectors.toList());
}

Reply:
Sounds good.